spec/file_database_spec.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require 'file_database'
+require 'tempfile'
+
+RSpec.describe FileDatabase do
+  let(:file) { Tempfile.open('test') }
+  let(:database) { described_class.new(file) }
+
+  after(:each) { file.unlink }
+
+  describe '#all' do
+    it 'reads all contacts in a json file to ruby' do
+      create_json_contact
+
+      expect(database.all).to eq([test_details])
+    end
+
+    it 'return an empty array if file is empty' do
+      expect(database.all).to eq([])
```

Comment:
you can use RSpec :sparkles: magic :sparkles: and do `.to be_empty`

Reply:
will that still fail if it doesn't specificity, return an array?

Reply:
the code this is testing always returns an empty array if there are no contacts

if you wanted to return something different when there are no contacts you would have to change the test first, then change the code

Reply:
my point is, the requirement for the code is that it only returns an empty array.
.to be_empty returns true on empty strings, hashs and arrays, so the test will not be testing its description/specification.

Reply:
you can leave it as you have it too